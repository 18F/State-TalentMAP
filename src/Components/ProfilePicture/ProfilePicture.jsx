import React from 'react';
import PropTypes from 'prop-types';

const ProfilePicture = ({ alt, source, className }) => (
  <img
    alt={alt}
    className={`dashboard-user-profile-picture ${className}`}
    src={source}
  />
);

ProfilePicture.propTypes = {
  alt: PropTypes.string,
  className: PropTypes.string,
  source: PropTypes.string,
};

ProfilePicture.defaultProps = {
  alt: 'avatar',
  className: '',
  source: '/assets/img/avatar.png',
};

export default ProfilePicture;
