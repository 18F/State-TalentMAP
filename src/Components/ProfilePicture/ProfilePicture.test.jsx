import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ProfilePicture from './ProfilePicture';

describe('ProfilePicture', () => {
  it('is defined', () => {
    const wrapper = shallow(<ProfilePicture />);
    expect(wrapper).toBeDefined();
  });

  it('can accept an alt tag', () => {
    const alt = 'alt text';
    const wrapper = shallow(<ProfilePicture alt={alt} />);
    expect(wrapper.find('img').prop('alt')).toBe(alt);
    expect(wrapper).toBeDefined();
  });

  it('can pass an optional className prop', () => {
    const className = 'special-class';
    const wrapper = shallow(<ProfilePicture className={className} />);
    expect(wrapper.find(`.${className}`).exists()).toBe(true);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<ProfilePicture />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
