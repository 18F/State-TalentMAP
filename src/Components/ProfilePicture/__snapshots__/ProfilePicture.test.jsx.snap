// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture matches snapshot 1`] = `
<img
  alt="avatar"
  className="dashboard-user-profile-picture "
  src="/assets/img/avatar.png"
/>
`;
